bentoml/yatai/yatai_service_impl.py

```diff
@@ -356,23 +356,45 @@ def UpdateBento(self, request, context=None):
     def DangerouslyDeleteBento(self, request, context=None):
         try:
             # TODO: validate request
-            bento_pb = self.bento_metadata_store.get(
-                request.bento_name, request.bento_version
-            )
-            if not bento_pb:
-                msg = (
-                    f"BentoService {request.bento_name}:{request.bento_version} "
-                    f"has already been deleted"
-                )
-                return DangerouslyDeleteBentoResponse(status=Status.ABORTED(msg))
+            bento_pb_list_for_delete = []
+            if request.all:
+                # Delete all bentos
+                bento_pb_list_for_delete = self.bento_metadata_store.list()
+                self.bento_metadata_store.dangerously_batch_delete()
+            else:
+                if request.bento_name and request.bento_version:
+                    # Delete a single bento
+                    bento_pb = self.bento_metadata_store.get(
+                        request.bento_name, request.bento_version
+                    )
+                    if not bento_pb:
+                        msg = (
+                            f"BentoService "
+                            f"{request.bento_name}:{request.bento_version} has already "
+                            f"been deleted"
+                        )
+                        return DangerouslyDeleteBentoResponse(
+                            status=Status.ABORTED(msg)
+                        )
 
-            logger.debug(
-                'Deleting BentoService %s:%s', request.bento_name, request.bento_version
-            )
-            self.bento_metadata_store.dangerously_delete(
-                request.bento_name, request.bento_version
-            )
-            self.repo.dangerously_delete(request.bento_name, request.bento_version)
+                    self.bento_metadata_store.dangerously_delete(
+                        request.bento_name, request.bento_version
+                    )
+                    bento_pb_list_for_delete.append(bento_pb)
+                else:
+                    # Delete a list of bentos
+                    bento_pb_list_for_delete = self.bento_metadata_store.list(
+                        bento_name=request.bento_name,
+                        label_selectors=request.label_selectors,
+                    )
+                    self.bento_metadata_store.dangerously_batch_delete(
+                        bento_pb_list_for_delete
+                    )
+            for bento in bento_pb_list_for_delete:
```

Comment:
Why is this for loop still required here?

Reply:
This is for deleting the bundle files from storage.  Instead of implementing bulk delete for S3 and GCS, I opt to loop to delete them for now.